libopenage/curve/keyframe.h

```diff
@@ -27,16 +27,26 @@ class Keyframe {
 	 * New, default-constructed element at the given time
 	 */
 	Keyframe(const time::time_t &time) :
-		time{time} {}
+		timestamp{time} {}
 
 	/**
 	 * New element fron time and value
 	 */
 	Keyframe(const time::time_t &time, const T &value) :
-		time{time},
+		timestamp{time},
 		value{value} {}
 
-	const time::time_t time = time::TIME_MIN;
+	const time::time_t &time() const {
+		return this->timestamp;
+	}
+
+	const T &val() const {
+		return this->value;
+	}
+
+private:
+	time::time_t timestamp = time::TIME_MIN;
```

Comment:
why not leave the time as a const member? then no function to access it is required.

Reply:
If it is `const` then it is not copy assignable/constructible which becomes a problem for working with vector (or deque). I know I can write a new copy/assignment contructor but making them private looks more clean to me.